libqtile/layout/floating.py

```diff
@@ -181,31 +181,35 @@ def compute_client_position(self, client, screen_rect):
         """ recompute client.x and client.y, returning whether or not to place
         this client above other windows or not """
         above = False
-        transient_for = client.window.get_wm_transient_for()
-        win = client.group.qtile.windows_map.get(transient_for)
-        if win is not None:
-            # if transient for a window, place in the center of the window
-            center_x = win.x + win.width / 2
-            center_y = win.y + win.height / 2
-        else:
-            center_x = screen_rect.x + screen_rect.width / 2
-            center_y = screen_rect.y + screen_rect.height / 2
-            above = True
-
-        x = center_x - client.width / 2
-        y = center_y - client.height / 2
-
-        # don't go off the right...
-        x = min(x, screen_rect.x + screen_rect.width)
-        # or left...
-        x = max(x, screen_rect.x)
-        # or bottom...
-        y = min(y, screen_rect.y + screen_rect.height)
-        # or top
-        y = max(y, screen_rect.y)
-
-        client.x = int(round(x))
-        client.y = int(round(y))
+        if (client.x == 0 and client.y == 0) \
+                or (client.x + client.width < screen_rect.x) or (client.x > screen_rect.x + screen_rect.width) \
+                or (client.y + client.height < screen_rect.y) or (client.y > screen_rect.y + screen_rect.width):
+            # client has not been properly placed before or it is off screen
+            transient_for = client.window.get_wm_transient_for()
+            win = client.group.qtile.windows_map.get(transient_for)
+            if win is not None:
+                # if transient for a window, place in the center of the window
+                center_x = win.x + win.width / 2
+                center_y = win.y + win.height / 2
+            else:
+                center_x = screen_rect.x + screen_rect.width / 2
+                center_y = screen_rect.y + screen_rect.height / 2
+                above = True
+
+            x = center_x - client.width / 2
+            y = center_y - client.height / 2
+
+            # don't go off the right...
+            x = min(x, screen_rect.x + screen_rect.width - client.width)
```

Comment:
I made a small change here, since it seems more sensible to have the entire window on screen than just requiring it to be adjacent to it.

Reply:
Can you add this reasoning to the commit message itself?

Reply:
OK, I reworded the commit message.